kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/base/BaseRecyclerAdapter.java

```diff
@@ -0,0 +1,181 @@
+package com.kin.ecosystem.base;
+
+import android.content.Context;
+import android.support.annotation.IntRange;
+import android.support.annotation.LayoutRes;
+import android.support.annotation.Nullable;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.FrameLayout;
+import java.util.ArrayList;
+import java.util.List;
+
+public abstract class BaseRecyclerAdapter<T, VH extends BaseViewHolder> extends RecyclerView.Adapter<VH> {
+
+	private static final int EMPTY_VIEW = 0x00000001;
+
+	private OnItemClickListener onItemClickListener;
+	private LayoutInflater layoutInflater;
+	private @LayoutRes int layoutResId;
+
+	private boolean isEmptyViewVisible;
+	private FrameLayout emptyLayout;
+	private RecyclerView recyclerView;
+
+	private List<T> data;
+
+	public BaseRecyclerAdapter(@LayoutRes int layoutResId) {
+		this.layoutResId = layoutResId;
+	}
+
+	protected abstract void convert(VH holder, @Nullable T item);
+
+	protected abstract VH createBaseViewHolder(View view);
+
+	public void setNewData(@Nullable List<T> data) {
+		this.data = data == null ? new ArrayList<T>() : data;
+		this.isEmptyViewVisible = false;
```

Comment:
if data == null should this.isEmptyViewVisible = false;?

Reply:
if this.data equal data should we call notifyDataSetChanged()?

Reply:
will fo over this file later

Reply:
this could have ben a separate PR

Reply:
1. yes -> notifydatasetchange will handle it.
2. this is not the adapter responsibility to handle this scenario.
3. k
4. yup